src/components/Avatar/Avatar.tsx

```diff
@@ -0,0 +1,70 @@
+/** @jsx jsx */
+import { jsx } from '@emotion/core';
+import React from 'react';
+import { avatarStyle } from './Avatar.style';
+import { pickTextColorFromSwatches } from '../../theme/palette';
+import { AcceptedIconNames } from '../Icon/types';
+import Icon from '../Icon';
+import { calculateActualColorFromComponentProp } from '../../utils/themeFunctions';
+
+export type Props = {
+  /** the src of the image to show **/
+  src?: string;
+  /** The icon name to pick from our library of icons
+   *  @default 'user'
+   * */
+  iconName?: AcceptedIconNames;
+  /** The size of the avatar
+   *  @default 'md'
+   * */
+  size?: AvatarSizes;
+  /** the color of the button based on our colors eg. red-400
+   *  @default 'darkGray-400'
+   * */
+  color?: string;
+  /** the shape of the avatar
+   *  @default 'circular'
+   * */
+  shape?: AvatarShapes;
+  /** The class name of the avatar component if its styled **/
+  className?: string;
+};
+
+export type AvatarSizes = 'xs' | 'sm' | 'md' | 'lg';
+export type AvatarShapes = 'regular' | 'rounded' | 'circular';
+
+const Avatar: React.FC<Props> = ({
+  src = '',
+  iconName = 'user',
+  size = 'md',
+  color = 'lightGray-600',
+  shape = 'circular',
+  children,
+  className,
+}) => {
+  const calculatedColor = calculateActualColorFromComponentProp(color);
+
+  return (
+    <div
+      className={className}
+      css={avatarStyle({
+        shape,
+        size,
+        fill: calculatedColor.color,
+        fillShade: calculatedColor.shade,
+      })}
+    >
+      {src && <img src={src} />}
+      {!src && !children && iconName && (
```

Comment:
Do we have the same case here as the TextField? Should we let the user decide how to pass his icon/image?

Also, give an alt to the image 😄

Reply:
this is an `iconName` for sure. It will never be an external icon here unless its a src of an avatar